src/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -395,23 +394,13 @@ public ThreadPoolWorkQueue()
 
         internal void EnsureThreadRequested()
         {
-            //
-            // If we have not yet requested #procs threads from the VM, then request a new thread
-            // as needed
+            // If we have not yet a thread request outstanding from the VM, then request a new thread
             //
             // Note that there is a separate count in the VM which will also be incremented in this case, 
             // which is handled by RequestWorkerThread.
-            //
-            int count = numOutstandingThreadRequests;
-            while (count < ThreadPoolGlobals.processorCount)
+            if (Interlocked.Exchange(ref threadRequestOutstanding, 1) == 0)
             {
-                int prev = Interlocked.CompareExchange(ref numOutstandingThreadRequests, count + 1, count);
-                if (prev == count)
-                {
-                    ThreadPool.RequestWorkerThread();
-                    break;
-                }
-                count = prev;
+                ThreadPool.RequestWorkerThread();
```

Comment:
> An issue with this is going to be that when an idle thread pool gets a burst of work items in quick succession, only one thread would be requested, and only one thread then will be available to process all of that work.

Except... one of the things we were trying to fix in the reverted PR https://github.com/dotnet/coreclr/issues/14017 was the over requesting of threads; because the very first thing `Dispatch` if it dequeues a work item; before running it, is to request another thread (as does Enqueue - so double requesting):

https://github.com/dotnet/coreclr/blob/208ea16c870cc2b60682d721bd4ce78db14a2805/src/System.Private.CoreLib/src/System/Threading/ThreadPool.cs#L561-L564

So it will always (except in gap) have a request active while it is successfully dequeueing; though only 1 rather than # cpu count requests active?

Reply:
Rather than trying to fix that, which was problematic and difficult to get round the races; idea is to work with it. So it will always (except between wake-up and dequeue gap) have a request active while it is successfully dequeueing; though only 1 rather than # cpu count?

Reply:
I believe the over-requesting is currently necessary because the thread request count tracking is fragile, mainly to reduce the chance of failure to near-zero (but not zero), theoretically. @sdmaclea tried to limit the requesting to the right amount but it ended up having races - I don't think there was anything wrong with the solution, but because of existing issues, it didn't work.

Reply:
Agreed, but this takes a different approach by instead working with the over-requesting? 

So with this change, while there is any items in the queue there will always[1] be **one** thread request active (plus when there are no more items, but one has just been dequeued).

Whereas currently it will always very quickly grow to always **ProcessorCount** thread requests active (as the number will never decrease while items are dequeued); as well as having a hot CompareExchange loop to decrease and increase the number.

[1] There will be a slight gap between thread being activated and dequeing an item, where it will drop to 0